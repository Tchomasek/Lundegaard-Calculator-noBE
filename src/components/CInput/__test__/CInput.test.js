import { CInput } from "../index";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <CInput
      info={""}
      onSubmit={() => {}}
      value={0}
      min_value={0}
      max_value={1}
      onChangeSlider={() => {}}
      marks={[]}
      error={false}
      helperText={""}
      displayedValue={0}
      onChangeTextInput={() => {}}
      onBlur={() => {}}
      unit={""}
    />,
    div
  );
});

it("renders button correctly", () => {
  const { getByTestId } = render(
    <CInput
      info={"Info"}
      onSubmit={() => {}}
      value={0}
      min_value={0}
      max_value={1}
      onChangeSlider={() => {}}
      marks={[]}
      error={false}
      helperText={""}
      displayedValue={0}
      onChangeTextInput={() => {}}
      onBlur={() => {}}
      unit={""}
    />
  );
  expect(getByTestId("CInput")).toHaveTextContent("Info");
});

it("matches snapshot", () => {
  const button = renderer.create(<CInput info={"Click"} />).toJSON();
  expect(button).toMatchSnapshot();
});
