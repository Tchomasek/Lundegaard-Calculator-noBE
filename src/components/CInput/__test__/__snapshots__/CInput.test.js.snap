// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-row-1 MuiGrid-container"
  data-testid="CInput"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    Click
  </h6>
  <div
    className="MuiGrid-root makeStyles-input-3 MuiGrid-container"
  >
    <span
      className="MuiSlider-root MuiSlider-colorPrimary makeStyles-slider-2"
      onMouseDown={[Function]}
    >
      <span
        className="MuiSlider-rail"
      />
      <span
        className="MuiSlider-track"
        style={
          Object {
            "left": "0%",
            "width": "NaN%",
          }
        }
      />
      <input
        type="hidden"
        value="NaN"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={NaN}
        className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="slider"
        style={
          Object {
            "left": "NaN%",
          }
        }
        tabIndex={0}
      />
    </span>
    <div
      className="MuiGrid-root makeStyles-inputWithUnit-4"
    >
      <form>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textField-5"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-13"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
      <p
        className="MuiTypography-root makeStyles-unit-6 MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
